mkdocs/generate_code_reference.py

```diff
@@ -29,12 +29,55 @@ def source_file_generator(src_paths):
 
     nav[parts] = doc_path.as_posix()
 
-    with mkdocs_gen_files.open(full_doc_path, "w") as fd:
-        if parts[-1] == "__init__":
-            ident = ".".join(parts[:-1])
+    def get_reference_file_string(p):
+        """
+
+        Args:
+            p: a tuple of module names which combine to make an identifier
+
+        Returns:
+            A markdown string with the full combined module name (excluding any __init__ part) as
+            the L1-title and the mkdocstrings "::: <identifier>" below
+
+        Examples:
+            >>> print(get_reference_file_string(("a",)))
+            # a
+            <BLANKLINE>
+            ::: a
+
+            >>> print(get_reference_file_string(("__init__",)))
+            Traceback (most recent call last):
+            ...
+            AssertionError: __init__ must have a parent
+
+            >>> print(get_reference_file_string(("a", "b")))
+            # a.b
+            <BLANKLINE>
+            ::: a.b
+
+            >>> print(get_reference_file_string(("a", "b", "__init__")))
+            # a.b
+            <BLANKLINE>
+            ::: a.b.__init__
+
+                        >>> print(get_reference_file_string(("a", "b")))
```

Comment:
for my own instruction, what is this last example adding that isn't in the one in lines 53-56 above?. i see the extra indentation, but i don't understand what is incremental. thanks!

Reply:
It's broken! Thanks for catching that, I'll take it out.